File: trunk/quote/Projects/PluginModel/quote/PrimaryPropeties.vb
Imports System.ComponentModel
Imports System.Reflection

Imports Model.Template
Imports Model

Namespace Quote

    Public Class PrimaryProperties
        Inherits Common.PrimaryProperties

        Private _RequestForQuoteNumber As String
        Private _PartNumber As String
        Private _TemplateID As Long
        Private _Initials As String
        Private _CreatedDate As DateTime
        Private _LastModified As DateTime

        Public Sub New(ByVal id As Long, _
                       ByVal requestForQuoteNumber As String, _
                       ByVal partNumber As String, _
                       ByVal initials As String, _
                       ByVal createdDate As DateTime, _
                       ByVal lastModified As DateTime
)
            Me.SetId(id)
            Me._RequestForQuoteNumber = requestForQuoteNumber
            Me._PartNumber = partNumber
            Me._Initials = initials
            Me._CreatedDate = createdDate
            Me._LastModified = lastModified
        End Sub

        <CategoryAttribute(Spaces.SortedSpaces1 + "Date"), _
         DisplayName("CreatedDate"), _
         DescriptionAttribute("Created Date")> _
        Public ReadOnly Property CreatedDate As DateTime
            Get
                Return _CreatedDate
            End Get
        End Property

        <CategoryAttribute(Spaces.SortedSpaces1 + "Date"), _
         DisplayName("LastModified"), _
            DescriptionAttribute("Last Modified Date")> _
        Public ReadOnly Property LastModified As DateTime
            Get
                Return _LastModified
            End Get
        End Property

        <CategoryAttribute(Spaces.SortedSpaces2 + "Misc"), _
        DisplayName("Quote Number"), _
        DescriptionAttribute("Quote Number")> _
        Public Overloads ReadOnly Property QuoteNumber As Integer
            Get
                Return MyBase.CommonId
            End Get
        End Property

        <CategoryAttribute(Spaces.SortedSpaces2 + "Misc"), _
        DisplayName("Initials"), _
        DescriptionAttribute("Initials of creator")> _
        Public ReadOnly Property Initials As String
            Get
                Return _Initials
            End Get
        End Property

        <CategoryAttribute(Spaces.SortedSpaces3 + "Quote"), _
        DisplayName("Customer"), _
        DescriptionAttribute("The customer"),
        TypeConverter(GetType(CustomerConverter))> _
        Public ReadOnly Property Customer As Customer
            Get
                Return Me.CommonCustomer
            End Get
        End Property

        <CategoryAttribute(Spaces.SortedSpaces3 + "Quote"), _
        DisplayName("Part Number"), _
        DescriptionAttribute("Part Number")> _
        Public ReadOnly Property PartNumber As String
            Get
                Return _PartNumber
            End Get
        End Property

        <CategoryAttribute(Spaces.SortedSpaces3 + "Quote"), _
        DisplayName("RFQ"), _
        DescriptionAttribute("Request For Quote")> _
        Public ReadOnly Property RequestForQuoteNumber As String
            Get
                Return _RequestForQuoteNumber
            End Get
        End Property

        <CategoryAttribute(Spaces.SortedSpaces3 + "Quote"), _
        DisplayName("TemplateNumber"), _
        DescriptionAttribute("Created from template")> _
        Public ReadOnly Property TemplateNumber As Integer
            Get
                Return Me._TemplateID
            End Get
        End Property

        Public Sub SetTemplateId(ByVal id As Long)
            Me._TemplateID = id
        End Sub

    End Class
End Namespace

File: quote/Projects/PluginModel/quote/PrimaryPropeties.cs
namespace Model.Quote
{
    using System;
    using System.ComponentModel;
    using System.Linq;
    
    using Model.Template;

    /// <summary>
    /// The primary properties of a quote.
    /// </summary>
    public class PrimaryProperties : Common.PrimaryProperties
    {
        /// <summary>
        /// The request for quote number.
        /// </summary>
        private string requestForQuoteNumber;

        /// <summary>
        /// The part number of the quote.
        /// </summary>
        private string partNumber;

        /// <summary>
        /// The template id.
        /// </summary>
        private int templateID;
        
        /// <summary>
        /// The initials of the person who created the quote.
        /// </summary>
        private string initials;
        
        /// <summary>
        /// The create date of the quote.
        /// </summary>
        private DateTime createdDate;

        /// <summary>
        /// The last modified date of the quote.
        /// </summary>
        private DateTime lastModified;
        
        /// <summary>
        /// Initializes a new instance of the <see cref="PrimaryProperties" /> class.
        /// </summary>
        /// <param name="id">The quote id.</param>
        /// <param name="requestForQuoteNumber">The request for quote number.</param>
        /// <param name="partNumber">The part number.</param>
        /// <param name="initials">The initials.</param>
        /// <param name="createdDate">The create date.</param>
        /// <param name="lastModified">The last modified date.</param>
        /// <param name="templateId">The template id.</param>
        public PrimaryProperties(
            int id, 
            string requestForQuoteNumber, 
            string partNumber, 
            string initials, 
            DateTime createdDate, 
            DateTime lastModified,
            int templateId)
        {
            this.SetId(id);
            this.requestForQuoteNumber = requestForQuoteNumber;
            this.partNumber = partNumber;
            this.initials = initials;
            this.createdDate = createdDate;
            this.lastModified = lastModified;
            this.templateID = templateId;
        }

        /// <summary>
        /// Gets the create date.
        /// </summary>
        /// <value>The create date.</value>
        [CategoryAttribute(Spaces.SortedSpaces1 + "Date"), DisplayName("CreatedDate"), DescriptionAttribute("Created Date")]
        public DateTime CreatedDate 
        {
            get { return this.createdDate; }
        }

        /// <summary>
        /// Gets the last modified date.
        /// </summary>
        /// <value>The last modified date.</value>
        [CategoryAttribute(Spaces.SortedSpaces1 + "Date"), DisplayName("LastModified"), DescriptionAttribute("Last Modified Date")]
        public DateTime LastModified 
        {
            get { return this.lastModified; }
        }

        /// <summary>
        /// Gets the quote number.
        /// </summary>
        /// <value>The quote number.</value>
        [CategoryAttribute(Spaces.SortedSpaces2 + "Misc"), DisplayName("Quote Number"), DescriptionAttribute("Quote Number")]
        public int QuoteNumber 
        {
            get { return this.CommonId; }
        }

        /// <summary>
        /// Gets the initials of the person who created the quote.
        /// </summary>
        /// <value>The initials.</value>
        [CategoryAttribute(Spaces.SortedSpaces2 + "Misc"), DisplayName("Initials"), DescriptionAttribute("Initials of creator")]
        public string Initials 
        {
            get { return this.initials; }
        }

        /// <summary>
        /// Gets the customer object.
        /// </summary>
        /// <value>The customer object.</value>
        [CategoryAttribute(Spaces.SortedSpaces3 + "Quote"), DisplayName("Customer"), DescriptionAttribute("The customer"), TypeConverter(typeof(CustomerConverter))]
        public Customer Customer 
        {
            get { return this.CommonCustomer; }
        }

        /// <summary>
        /// Gets the part number.
        /// </summary>
        /// <value>The part number.</value>
        [CategoryAttribute(Spaces.SortedSpaces3 + "Quote"), DisplayName("Part Number"), DescriptionAttribute("Part Number")]
        public string PartNumber 
        {
            get { return this.partNumber; }
        }

        /// <summary>
        /// Gets the request for quote number.
        /// </summary>
        /// <value>The request for quote number.</value>
        [CategoryAttribute(Spaces.SortedSpaces3 + "Quote"), DisplayName("RFQ"), DescriptionAttribute("Request For Quote")]
        public string RequestForQuoteNumber 
        {
            get { return this.requestForQuoteNumber; }
        }

        /// <summary>
        /// Gets the template number.
        /// </summary>
        /// <value>The template number.</value>
        [CategoryAttribute(Spaces.SortedSpaces3 + "Quote"), DisplayName("TemplateNumber"), DescriptionAttribute("Created from template")]
        public int TemplateNumber 
        {
            get { return this.templateID; }
        }

        /// <summary>
        /// Sets the template id.
        /// </summary>
        /// <param name="id">The template id.</param>
        public void SetTemplateId(int id)
        {
            this.templateID = id;
        }
    }
}
